src/kvstore/wal/FileBasedWalIterator.cpp

```diff
@@ -240,75 +176,69 @@ LogID FileBasedWalIterator::logId() const {
     return currId_;
 }
 
-
+// use logEntry to retrieve wal log
 TermID FileBasedWalIterator::logTerm() const {
-    return currTerm_;
+    LOG(FATAL) << "Should not use this method";
 }
 
 
 ClusterID FileBasedWalIterator::logSource() const {
-    if (currId_ >= firstIdInBuffer_) {
-        // Retrieve from the buffer
-        DCHECK(!buffers_.empty());
-        return buffers_.front()->getCluster(currIdx_);
-    } else {
-        // Retrieve from the file
-        DCHECK(!fds_.empty());
+    LOG(FATAL) << "Should not use this method";
+}
 
-        ClusterID cluster = 0;
-        CHECK_EQ(pread(fds_.front(),
-                       &(cluster),
-                       sizeof(ClusterID),
-                       currPos_
-                        + sizeof(LogID)
-                        + sizeof(TermID)
-                        + sizeof(int32_t)),
-                 static_cast<ssize_t>(sizeof(ClusterID)))
-            << "Failed to read. Curr position is " << currPos_
-            << ", expected read length is " << sizeof(ClusterID)
-            << " (errno: " << errno << "): " << strerror(errno);
 
-        return cluster;
-    }
+folly::StringPiece FileBasedWalIterator::logMsg() const {
+    LOG(FATAL) << "Should not use this method";
 }
 
+LogEntry FileBasedWalIterator::logEntry() {
+    LogEntry logEntry;
+    // Try to get log from ring buffer first, if failed to read from buffer,
+    // try to read from wal files
+    // TODO: need to figure out when can we stop reading from wal file anymore once hit ring buffer
+    if (!ringBuffer_->getLogEntry(currId_, logEntry)) {
+        if (!needToReadWalFile_) {
+            loadWalFiles();
+            needToReadWalFile_ = true;
```

Comment:
change needToReadWalFile_  to hasBeenReadWalFile_?

Reply:
The variable is false typically because we read log from buffer. If it is out of ring buffer, then we need to read from wal files, so I think needToReadWalFile_ make more sense

Reply:
needToReadWalFile_  to chinese is "是否需要读wal", when it is false, "不需要读wal", but the logic is read wal file.

Reply:
Fair enough